pkg/config/config.go

```diff
@@ -2,35 +2,52 @@ package config
 
 import (
 	"fmt"
-	"github.com/redhat-developer/odo/pkg/util"
 	"io/ioutil"
 	"os"
 	"os/user"
 	"path/filepath"
+	"reflect"
 	"strconv"
 	"strings"
 
+	"github.com/redhat-developer/odo/pkg/util"
+
 	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 )
 
 const (
-	configEnvName  = "ODOCONFIG"
-	configFileName = "odo"
+	globalConfigEnvName = "GLOBALODOCONFIG"
+	localConfigEnvName  = "LOCALODOCONFIG"
+	configFileName      = "odo-config.yaml"
 	//DefaultTimeout for openshift server connection check
 	DefaultTimeout = 1
 )
 
+// Info is implemented by configuration managers
+type Info interface {
+	SetConfiguration(parameter string, value string) error
+	GetConfiguration(parameter string) (interface{}, bool)
+	DeleteConfiguration(parameter string) error
+}
+
 // OdoSettings holds all odo specific configurations
 type OdoSettings struct {
 	// Controls if an update notification is shown or not
-	UpdateNotification *bool `json:"updatenotification,omitempty"`
+	UpdateNotification *bool `json:"updateNotification,omitempty"`
```

Comment:
Maybe we can have everything "CamelCase" instead of "camelCase" ("UpdateNotification")

Reply:
ack

Reply:
resolved

Reply:
resolved